python/cudf/cudf/core/multiindex.py

```diff
@@ -1479,21 +1405,14 @@ def from_pandas(cls, multiindex, nan_as_null=None):
         # which preserves all levels of `multiindex`.
         names = tuple(range(len(multiindex.names)))
 
-        mi = cls(
-            names=multiindex.names,
-            source_data=multiindex.to_frame(name=names),
-            nan_as_null=nan_as_null,
+        df = cudf.DataFrame.from_pandas(
+            multiindex.to_frame(index=False, name=names), nan_as_null
         )
+        return cls.from_frame(df, names=multiindex.names)
 
-        return mi
-
-    @property
+    @cached_property
     def is_unique(self):
-        if not hasattr(self, "_is_unique"):
-            self._is_unique = len(self._source_data) == len(
-                self._source_data.drop_duplicates(ignore_index=True)
-            )
-        return self._is_unique
+        return len(self) == len(self.unique())
```

Comment:
```suggestion
        return len(self) == len(self.drop_duplicates(ignore_index=True))
```

We can ignore the index here.

Reply:
Also, any reason we want to drop caching this property?

Reply:
Should I instead change the definition of `unique`? We don't want to consider the index there either, right?

The property is still being cached, it's just a little harder to see in the Github diff. If you look above the `def` I've changed the decorator from `property` to `cached_property` so that we don't have to reproduce caching logic here, that's all.

Reply:
Discussed offline, @galipremsagar agreed with changing the behavior of `unique`.